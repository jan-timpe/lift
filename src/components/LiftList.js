import React, { Component } from 'react'
import '../style/LiftList.css'

import Lift from './Lift'

export default class LiftList extends Component {

  render() {
    return (
      <div className="LiftList">
        {this.props.lifts.map((lift, index) => (
          <Lift lift={lift} key={index} />
        ))}
      </div>
    )
  }
}