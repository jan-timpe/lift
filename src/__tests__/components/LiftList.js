import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LiftList from '../../components/LiftList'


describe(`LiftList`, () => {

  // snapshot
  it(`should render`, () => {
    const lifts = [{
      id: 0,
      name: 'Squat'
    }, {
      id: 1,
      name: 'Bench'
    }]
    const wrapper = shallow(<LiftList lifts={lifts} />)
    const component = wrapper
    expect(toJson(component)).toMatchSnapshot()
  })

})