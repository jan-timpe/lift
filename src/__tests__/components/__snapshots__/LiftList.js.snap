// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiftList should render 1`] = `
<div
  className="LiftList"
>
  <Connect(Lift)
    key="0"
    lift={
      Object {
        "id": 0,
        "name": "Squat",
      }
    }
  />
  <Connect(Lift)
    key="1"
    lift={
      Object {
        "id": 1,
        "name": "Bench",
      }
    }
  />
</div>
`;
